packages/eslint-plugin/src/rules/explicit-function-return-type.ts

```diff
@@ -217,21 +217,20 @@ export default util.createRule<Options, MessageIds>({
         return false;
       }
 
-      // Check if body is a block with a single statement
-      if (
-        body.type === AST_NODE_TYPES.BlockStatement &&
-        body.body.length === 1
-      ) {
-        const [statement] = body.body;
-
-        // Check if that statement is a return statement with an argument
-        if (
-          statement.type === AST_NODE_TYPES.ReturnStatement &&
-          !!statement.argument
-        ) {
-          // If so, check that returned argument as body
-          body = statement.argument;
-        }
+      // Check if body is a block with a return statement
+      if (body.type === AST_NODE_TYPES.BlockStatement) {
+        // Check if ReturnStatement exist among body and use that argument
+        // for checking whether it is a function expression
+        body.body.some(statement => {
```

Comment:
There's one problem with this code - it'll allow you to have multiple return statements nested in the code, which could have different return types.

ie - the following code will pass:
```ts
function foo() {
  if (true) {
    return 1;
  }

  return (): void => {};
}
```

This is where things get difficult. The best way to approach this is via maintaining state during traversal and then doing checks on the `:exit`.
Eg https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/rules/require-await.ts

Reply:
Yeah you are definitely right, I'll make a fix(though I don't have much time right now so it'll be done probably after few weeks...)